### PR TITLE
add strip_solution tests

### PR DESCRIPTION
**Pull request by jClugstor**
## Checklist

- [ x] Appropriate tests were added
- [ x] Any code changes were done in a way that does not break public API
- [ x] All documentation related to code changes were updated
- [ x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ x] Any new documentation only uses public API

### CODE REVIEW

File: test/interface/ode_strip_test.jl [hunk 1]
```diff
@@ -15,3 +15,10 @@ sol = solve(prob, Rosenbrock23())
```
Comment (ChrisRackauckas) [inline test/interface/ode_strip_test.jl:15]: also test an implicit method, `TRBDF2`. Also test that a lazy method does the right thing.

File: test/interface/ode_strip_test.jl [hunk 2]
```diff
@@ -11,14 +11,30 @@ u0 = [1.0; 0.0; 0.0]
...
-@test isnothing(SciMLBase.strip_interpolation(sol.interp).cache.jac_config)
-@test isnothing(SciMLBase.strip_interpolation(sol.interp).cache.grad_config)
 
-@test isnothing(SciMLBase.strip_solution(sol).prob)
-@test isnothing(SciMLBase.strip_solution(sol).alg)
-@test isnothing(SciMLBase.strip_solution(sol).interp.f)
-@test isnothing(SciMLBase.strip_solution(sol).interp.cache.jac_config)
-@test isnothing(SciMLBase.strip_solution(sol).interp.cache.grad_config)
\ No newline at end of file
+@test_throws ErrorException SciMLBase.strip_solution(vern_sol)
```
Comment (ChrisRackauckas) [inline test/interface/ode_strip_test.jl:40]: It shouldn't be a generic error?

Comment (jClugstor) [reply test/interface/ode_strip_test.jl:40]: What kind of error should this be?

**Comment by ChrisRackauckas** [comment]
You shouldn't need to export anything. If you overload it properly (like what you show) that should be fine.

File: test/interface/ode_strip_test.jl (hunk 2 above)
Comment (ChrisRackauckas) [reply test/interface/ode_strip_test.jl:40]: LazyInterpolationException or something, Make a custom one so the GUI can catch it.

**Comment by ChrisRackauckas** [comment]
Test failure

File: test/interface/ode_strip_test.jl [hunk 3]
```diff
@@ -11,7 +11,30 @@ u0 = [1.0; 0.0; 0.0]
...
+    @test isnothing(SciMLBase.strip_solution(rosenbrock_sol).interp.cache.jac_config)
+    @test isnothing(SciMLBase.strip_solution(rosenbrock_sol).interp.cache.grad_config)
+end 
+
+@testset "TRBDF Solution Stripping" begin 
+    @test isnothing(SciMLBase.strip_solution(TRBDF_sol).prob)
+    @test isnothing(SciMLBase.strip_solution(TRBDF_sol).alg)
+    @test isnothing(SciMLBase.strip_solution(TRBDF_sol).interp.f)
+    @test isnothing(SciMLBase.strip_solution(TRBDF_sol).interp.cache.jac_config)
+    @test isnothing(SciMLBase.strip_solution(TRBDF_sol).interp.cache.grad_config)
```
Comment (ChrisRackauckas) [inline test/interface/ode_strip_test.jl:36]:
```suggestion
```

**Comment by ChrisRackauckas** [comment]
For TRBDF2, check the nlsolver is stripped https://github.com/SciML/OrdinaryDiffEq.jl/blob/master/lib/OrdinaryDiffEqSDIRK/src/sdirk_caches.jl#L15

File: test/interface/ode_strip_test.jl [hunk 4]
```diff
@@ -32,7 +32,6 @@ end
     @test isnothing(SciMLBase.strip_solution(TRBDF_sol).prob)
     @test isnothing(SciMLBase.strip_solution(TRBDF_sol).alg)
     @test isnothing(SciMLBase.strip_solution(TRBDF_sol).interp.f)
-    @test isnothing(SciMLBase.strip_solution(TRBDF_sol).interp.cache.jac_config)
```
[the remark this answers is not in the document]
Comment (ChrisRackauckas) [reply test/interface/ode_strip_test.jl]: ahh because nlsolver

[the remark this answers is not in the document]
Comment (jClugstor) [reply test/interface/ode_strip_test.jl]: Yeah there's no `jac_config`

**Comment by jClugstor** [comment]
Do I need to set `nlsolver` to `nothing` for any cache that has it as a field?

**Comment by ChrisRackauckas** [comment]
yes

**Pull request merged**